### PR TITLE
Rework on identifying & getting narrow from URL.

### PR DESCRIPTION
**Pull request by jainkuniya**
fixing https://sentry.io/zulip/zulip-mobile/issues/415961375/

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "object-curly-newline": 0,
     "no-bitwise": 0,
     "no-continue": 0,
+    "no-cond-assign": 0,
```
Comment (borisyankov) [inline .eslintrc]: You should not disable this.

File: src/utils/string.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* @flow */
+
+export const replaceAtIndex = (str: string, index: number, char: string): string =>
```
Comment (borisyankov) [inline src/utils/string.js]: You don't need this :)

File: src/utils/string.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  `${str.substring(0, index)}${char}${str.substring(index + 1)}`;
+
+export const transformToEncodedURI = (string: string): string => {
+  const pattern = /\.\d/g;
+  let match;
+  while ((match = pattern.exec(string))) {
```
Comment (borisyankov) [inline src/utils/string.js]: Have a look at:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

The `replace` function can receive a regexp, which will make the code way simpler.

File: .eslintrc (hunk 1 above)
Comment (jainkuniya) [reply .eslintrc]: Done 👍

File: src/utils/string.js (hunk 2 above)
Comment (jainkuniya) [reply src/utils/string.js]: Done 👍

File: src/utils/string.js (hunk 3 above)
Comment (jainkuniya) [reply src/utils/string.js]: Done 👍

**Comment by jainkuniya** [comment]
Updated 
ready to go in 🏃

**Comment by borisyankov** [comment]
Thanks. Good job!

**Pull request merged**